app/controllers/welcome_controller.rb

```diff
@@ -9,8 +9,28 @@ def new
       flash[:error] = "Already logged in!"
       redirect_to root_path
     end
-
     @developer = Developer.new
     @contractor = Contractor.new
   end
+
+  def deactivate_account
+  end
+
+  def deactivate
+    if current_contractor && current_contractor.authenticate(params[:deactivate][:password])
+      current_contractor.update_attribute(:status, "inactive")
+      session[:contractor_id] = nil
+      flash[:notice] = "Account successfully deactivated."
+      redirect_to root_path
+    elsif current_developer && current_developer.authenticate(params[:deactivate][:password])
+      current_developer.update_attribute(:status, "inactive")
+      session[:developer_id] = nil
+      flash[:notice] = "Account successfully deactivated."
+      redirect_to root_path
+    else
+      flash.now[:error] = 'Invalid login data'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.